src/core/nixl_agent.cpp

```diff
@@ -126,6 +128,8 @@ nixlAgent::getPluginParams (const nixl_backend_t &type,
 
     // TODO: unify to uppercase/lowercase and do ltrim/rtrim for type
 
+    NIXL_LOCK_GUARD(data->lock);
```

Comment:
Same here - it seems that plugin manager locking is enough.

It might be required later in this function when backend engines are accessed.

Reply:
Yep, this one is also redundant.